shared/engine/index.js

```diff
@@ -293,9 +293,9 @@ export type SessionID = number
 export type SessionIDKey = string // used in our maps, really converted to a string key
 export type WaitingHandlerType = (waiting: boolean, method: string, sessionID: SessionID) => void
 export type ResponseType = {
-  cancel: (...args: Array<any>) => void,
-  result: (...args: Array<any>) => void,
-  error: (...args: Array<any>) => void,
+  cancel(...args: Array<any>): void,
```

Comment:
This one's tricky. [Flow now treats method syntax as covariant, while regular property syntax is invariant](https://flowtype.org/blog/2016/10/04/Property-Variance.html#covariant-by-default-method-types). This means the def here conflicts with `IncomingRequest` in `shared/engine/request.js` because it uses the method syntax. The syntax used for declaring methods needs to match for `IncomingRequest` to be treated as a supertype of `ResponseType`. Flow doesn't do a good job hinting at this yet, so we'll need to keep an eye out for similar cases in the future.